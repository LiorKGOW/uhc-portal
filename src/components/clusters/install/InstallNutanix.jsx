import React from 'react';
import { PageSection } from '@patternfly/react-core';

import Breadcrumbs from '../../common/Breadcrumbs';
import links from '../../../common/installLinks.mjs';
import { InstructionsChooserPageTitle } from '~/components/clusters/install/instructions/InstructionsChooserPageTitle';
import { InstructionsChooser } from '~/components/clusters/install/instructions/InstructionsChooser';
import { AppPage } from '~/components/App/AppPage';

const InstallNutanix = () => {
  const breadcrumbs = (
    <Breadcrumbs
      path={[
        { label: 'Clusters' },
        { label: 'Cluster Type', path: '/create' },
        { label: 'Nutanix AOS' },
      ]}
    />
  );

  return (
    <AppPage title="Install OpenShift 4 | Red Hat OpenShift Cluster Manager | Nutanix AOS">
      <InstructionsChooserPageTitle cloudName="Nutanix AOS" breadcrumbs={breadcrumbs} />
      <PageSection>
        <InstructionsChooser
          aiPageLink="/assisted-installer/clusters/~new"
          aiLearnMoreLink={links.INSTALL_ASSISTED_LEARN_MORE}
          ipiPageLink="/install/nutanix/installer-provisioned"
          ipiLearnMoreLink={links.INSTALL_NUTANIXIPI_GETTING_STARTED}
          upiPageLink=""
          agentBasedPageLink=""
          hideUPI
          recommend="ipi"
          providerSpecificFeatures={{
            ipi: [
              'Hosts controlled with Nutanix AOS Cloud Provider',
              'For connected or air-gapped/restricted networks',
            ],
          }}
          name="nutanix"
        />
      </PageSection>
    </AppPage>
  );
};

export default InstallNutanix;
