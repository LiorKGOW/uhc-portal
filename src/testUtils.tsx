import React from 'react';
import { Provider } from 'react-redux';
import { createStore, Store } from 'redux';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';

import { toHaveNoViolations, axe } from 'jest-axe';

import { createBrowserHistory } from 'history';
import { store } from './redux/store';
import { reduxReducers } from './redux/reducers';
import * as restrictedEnv from './restrictedEnv';

const history = createBrowserHistory();

export const TestWrapper = ({
  children,
  initialStore,
}: {
  children: React.ReactNode;
  initialStore?: Store;
}) => <Provider store={initialStore || store}>{children}</Provider>;

const renderWithState = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  initialState?: any, // setting to any because only a partial state structure may be sent
) => {
  const newStore = initialState ? createStore(reduxReducers(history), initialState) : store;

  const Wrapper = ({ children }: { children: React.ReactNode }) => (
    <TestWrapper initialStore={newStore}>{children}</TestWrapper>
  );
  return { ...render(ui, { wrapper: Wrapper, ...options }), user: userEvent.setup() };
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';

// Type not exported in the library
export type UserEventType = {
  type: (
    field: HTMLElement,
    typeValue: string,
    { initialSelectionStart }: { initialSelectionStart: number },
  ) => void;
  clear: (field: HTMLElement) => void;
};

export { renderWithState as render };

/* ***** Items outside of React Test Library ************ */
expect.extend(toHaveNoViolations);

export const checkAccessibility = async (container: HTMLElement | string) => {
  const results = await axe(container);
  expect(results).toHaveNoViolations();
};

export const insightsMock = () => {
  global.insights = {
    chrome: {
      ...global.insights.chrome,
      on: () => () => {},
      appNavClick: () => {},
      auth: {
        getUser: () => Promise.resolve({ data: {} }),
        getToken: () => Promise.resolve(),
        getOfflineToken: () => Promise.resolve({ data: { refresh_token: 'hello' } }),
      },
    },
  };
};

export const mockRestrictedEnv = () => {
  const mock = jest.spyOn(restrictedEnv, 'isRestrictedEnv');
  mock.mockReturnValue(false);
  return mock;
};
