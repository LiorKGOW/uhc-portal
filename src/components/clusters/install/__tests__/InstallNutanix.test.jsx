import React from 'react';
import { CompatRouter } from 'react-router-dom-v5-compat';
import { screen, checkAccessibility, TestRouter, render } from '~/testUtils';

import InstallNutanix from '../InstallNutanix';

describe('InstallNutanix', () => {
  it('is accessible', async () => {
    const { container } = render(
      <TestRouter>
        <CompatRouter>
          <InstallNutanix />
        </CompatRouter>
      </TestRouter>,
    );

    expect(await screen.findByText('Create an OpenShift Cluster: Nutanix AOS')).toBeInTheDocument();
    await checkAccessibility(container);
  });
});
