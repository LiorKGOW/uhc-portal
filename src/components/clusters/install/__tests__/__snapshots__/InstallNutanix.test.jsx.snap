// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallNutanix renders correctly 1`] = `
<Fragment>
  <InstructionsChooserPageTitle
    breadcrumbs={
      <Breadcrumbs
        path={
          Array [
            Object {
              "label": "Clusters",
            },
            Object {
              "label": "Cluster Type",
              "path": "/create",
            },
            Object {
              "label": "Nutanix AOS",
            },
          ]
        }
      />
    }
    cloudName="Nutanix AOS"
  />
  <PageSection>
    <InstructionsChooser
      agentBasedPageLink=""
      aiLearnMoreLink="https://docs.openshift.com/container-platform/4.12/installing/installing_on_prem_assisted/installing-on-prem-assisted.html"
      aiPageLink="/assisted-installer/clusters/~new"
      hideUPI={true}
      ipiLearnMoreLink="https://docs.openshift.com/container-platform/4.12/installing/installing_nutanix/preparing-to-install-on-nutanix.html"
      ipiPageLink="/install/nutanix/installer-provisioned"
      name="nutanix"
      providerSpecificFeatures={
        Object {
          "ipi": Array [
            "Hosts controlled with Nutanix AOS Cloud Provider",
            "For connected or air-gapped/restricted networks",
          ],
        }
      }
      recommend="ipi"
      upiPageLink=""
    />
  </PageSection>
</Fragment>
`;
